### PR TITLE
ci: run SwiftLint on `mobile/**` changes

### PR DESCRIPTION
**Pull request by jpsim**
Commit Message: ci: run SwiftLint on `mobile/**` changes
Additional Description:
Risk Level: Low, CI job, only runs when on `mobile/**` changes
Testing: Local
Docs Changes: None
Release Notes: None
Platform Specific Features: only runs when on `mobile/**` changes
part of https://github.com/envoyproxy/envoy/pull/24224

### CODE REVIEW

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -5,10 +5,10 @@ on:
     branches:
       - main
     paths:
-    - 'mobile/**'
+      - 'mobile/**'
```
Comment (phlax) [inline .github/workflows/format.yml:9]: could you fix the linting here - we should be checking this but currently arent - there is #24197 pending to fix

**Comment by jpsim** [comment]
Confirmed SwiftLint job ran successfully when I forced it to run in https://github.com/envoyproxy/envoy/actions/runs/3565522319/jobs/5990798974

**Review by abeyad — APPROVED** [review]

**Comment by jpsim** [comment]
FYI I didn't do this, maybe some automation did? Repokitteh maybe? It's more aggressive than I'd like, I didn't need this PR to ping multiple people 🙃.

image

**Comment by alyssawilk** [comment]
Envoy codeowners results in auto-assigning reviewers.
in the Envoy repo we generally ignore reviewers and only review if a PR is assigned via the assignees field.

**Comment by abeyad** [comment]
> FYI I didn't do this, maybe some automation did? Repokitteh maybe? It's more aggressive than I'd like, I didn't need this PR to ping multiple people 🙃.
> 
> image

My understanding is that one of the github plugins automatically adds whoever is in the OWNERS set of the affected code to the "reviewers" list automatically.  But technically only the people in the "Assignees" list are responsible for the code review.

**Comment by alyssawilk** [comment]
want to just do a main merge so it'll run due to .workflow changes?

**Pull request merged**